Sources/PackageExtension/PackageExtension.swift

```diff
@@ -0,0 +1,339 @@
+        guard let idx = string.lastIndex(of: "/") else {
+            // No path separators, so the basename is the whole string.
+            return string
+        }
+        // Otherwise, it's the string from (but not including) the last path
+        // separator.
+        return String(string.suffix(from: string.index(after: idx)))
+    }
+    
+    public var suffix: String? {
+        // Find the last path separator, if any.
+        let sIdx = string.lastIndex(of: "/")
+        
+        // Find the start of the basename.
+        let bIdx = (sIdx != nil) ? string.index(after: sIdx!) : string.startIndex
+        
+        // Find the last `.` (if any), starting from the second character of
+        // the basename (a leading `.` does not make the whole path component
+        // a suffix).
+        let fIdx = string.index(bIdx, offsetBy: 1, limitedBy: string.endIndex) ?? string.startIndex
+        if let idx = string[fIdx...].lastIndex(of: ".") {
+            // Unless it's just a `.` at the end, we have found a suffix.
+            if string.distance(from: idx, to: string.endIndex) > 1 {
+                return String(string.suffix(from: string.index(idx, offsetBy: 1)))
+            }
+            else {
+                return nil
+            }
+        }
+        // If we get this far, there is no suffix.
+        return nil
+    }
+    
+    public func hasSuffix(_ suffix: String) -> Bool {
+        return self.string.hasSuffix(suffix)
+    }
+    
+    public init(stringLiteral value: String) {
+        self.init(value)
+    }
+
+    public init(extendedGraphemeClusterLiteral value: String) {
+        self.init(stringLiteral: value)
+    }
+
+    public init(unicodeScalarLiteral value: String) {
+        self.init(stringLiteral: value)
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        try container.encode(self.string)
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let string = try container.decode(String.self)
+        self.init(string)
+    }
+}
+
+
+
+/// -=-=-=-=-=-=-=-=-=-=-  Implementation Details Below  -=-=-=-=-=-=-=-=-=-=-
+
+
+
+/// Private initializer for the TargetBuildContext.  Expects the JSON form of
+/// this structure to be in the last command line argument.  Also registers a
+/// block to emit the output data at the end of the process.
+fileprivate func TargetBuildContextFactory() -> TargetBuildContext {
+    // Register a block to emit the JSON form of the output for SwiftPM to read.
+    atexit {
+        let encoder = JSONEncoder()
+        let data = try! encoder.encode(output)
+        fputc(0, stdout)
+        fputs(String(data: data, encoding: .utf8)!, stdout)
+        fflush(stdout)
+    }
+    // Look for the input JSON as the last argument of the invocation.
+    guard let data = ProcessInfo().arguments.last?.data(using: .utf8) else {
```

Comment:
This is something we want to fix.  It would probably be better to pass it in on stdin (to avoid any limits on command line lengths), but we don't have TSC support for that yet.